data/World/Shadow Temple MQ.json

```diff
@@ -153,7 +183,17 @@
         "region_name": "Shadow Temple Beyond Boat",
         "dungeon": "Shadow Temple",
         "locations": {
-            "Shadow Temple MQ GS After Ship": "Hookshot"
+            "Shadow Temple MQ GS After Ship": "Hookshot",
+            "Shadow Temple MQ After Boat Pot 1": "True",
+            "Shadow Temple MQ After Boat Pot 2": "True",
+            "Shadow Temple MQ After Boat Upper Recovery Heart 1" : "Bow and can_play(Song_of_Time) and Longshot",
+            "Shadow Temple MQ After Boat Upper Recovery Heart 2" : "Bow and can_play(Song_of_Time) and Longshot",
+            "Shadow Temple MQ After Boat Lower Recovery Heart" : "
+                (Bow or (logic_shadow_statue and has_bombchus) or shadow_temple_shortcuts)",
+            "Shadow Temple MQ Near Boss Pot 1": "
+                (Bow or (logic_shadow_statue and has_bombchus) or shadow_temple_shortcuts)",
+            "Shadow Temple MQ Near Boss Pot 2": "
+                (Bow or (logic_shadow_statue and has_bombchus) or shadow_temple_shortcuts)"
```

Comment:
What do the recovery hearts in this room share flags with? Honestly I can't remember if I've ever seen them before.